lbz/resource.py

```diff
@@ -207,6 +207,8 @@ def get_pagination(self, total_items: int, limit: int, offset: int) -> dict:
 
     @property
     def _pagination_uri(self) -> str:
-        if query_params := copy_without_keys(self.request.query_params, "offset", "limit"):
-            return f"{self.urn}?{urlencode(query_params)}&offset={{offset}}&limit={{limit}}"
+        if query_params := self.request.query_params.clean_keys(["offset", "limit"]):
```

Comment:
Removing these keys to achieve URL encoding makes sense but you cannot remove them permanently from the original `query_params` because they will be needed. I mean a situation like this:
```python
self.request.query_params["limit"]  # returns the limit provided from the outside
self._pagination_uri()  # returns URI but removes offset and limit
self.request.query_params["limit"]  # raises KeyError
```

You need to create a copy of those query parameters just to handle this case. Also, it would be good to cover it by the test, just to not break it in the future 😉